drivers/macvlan/macvlan_errors.go

```diff
@@ -0,0 +1,227 @@
+package macvlan
```

Comment:
We do not really need typed errors. Actually trying to move away from them. If you check overlay driver, it is not using them. You can just use the `types.<errors>` in case you need to return something the caller needs to evaluate, for example bridge driver to returns a MaskableError in some instances. One day we'll remove the typed errors from bridge driver as well.
(Also I see not all of the following are used in the code)